### DISCUSSION TITLE
How to monitor kafka service is ok when pod is running

### DISCUSSION THREAD

**Discussion by islet5** [strimzi/strimzi-kafka-operator#7626] [Q&A]
My kafka cluster is 0.20.0, might update it in the future. 
I followed doc to setup Prometheus to monitor it, created PodMonitor and PrometheusRule
<img width="1384" alt="image" src="https://user-images.githubusercontent.com/9824473/201842051-f0840d0e-926d-48cb-9582-806aa4ebfd2c.png">

<img width="1436" alt="image" src="https://user-images.githubusercontent.com/9824473/201842287-26dda2e5-58f8-411d-92bc-1d5ee2db96c4.png">

And is there any rule that i can monitor kafka service is ok when pod is in running state? Because last time i found pod is running but actually logs show certificate expired and kafka is not working.
![image](https://user-images.githubusercontent.com/9824473/201843017-a16ee950-0ceb-431c-9e7d-3e27fb2bb7f6.png)

**Comment by scholzj** [comment]
It is not clear which certificate is expired. So it is not really clear what exactly is not working.

**Reply by scholzj** [reply]
Also, it is not a binary state -> one client not being able to connect does not mean the cluster is not working and one client which can connect does not mean cluster is working. There are many _different_ states.

**Reply by islet5** [reply]
![image](https://user-images.githubusercontent.com/9824473/201944791-553df258-7dba-4975-bcfa-685688a871e0.png)

![image](https://user-images.githubusercontent.com/9824473/201947518-3d29d40c-1977-4c0e-9891-87695679e64a.png)


i can't tell which certificate is expired, i tried command as" kubectl -n dashboard get secret kafka-sensor-kafka-exporter-certs -o 'jsonpath={.data.kafka-exporter\.crt}' | base64 -d | openssl x509 -subject -issuer -startdate -enddate -noout" and checked some other secrets, didn't find any certificate is expired, guess they were renewed. I restarted pods by deleting them then error logs disappeared.

and why i think the cluster isn't working because developers find their services have log like this:
Error while fetching metadata with correlation id 715055 :

{gather=LEADER_NOT_AVAILABLE, jaeger=LEADER_NOT_AVAILABLE,sensorservice=LEADER_NOT_AVAILABLE}

**Reply by scholzj** [reply]
> i can't tell which certificate is expired

I think nobody can from the screenshots.

----

If the certificates are expired, you will probably need to recover from it manually:
* Delete all the secrets with the CAs and server certificates
* The operator will recreate them, but it will not be possible to roll the pods. So you will need to roll them manually (`kubectl delete pod`)
* Then (where needed) you will need to update the certificates in your clients etc.

0.20 is very old version which might have many different bugs.
